Bootstrap 4 compatibility fixes; Downloads fixed

SimSync opened the pull request:
**Bootstrap 4 compatibility**
- Several issues with Bootstrap 4 themes (e.g. Agency2) fixed
- Switched icons from glyphicons to fontawesome, because glyphicons are missing in Bootstrap 4

**Downloads**
- Switched links for the files list of a product, to use the vstore download function, to make sure that only "purchased" files are downloadable by the user.
- Fixed issue with the download url, as it invokes the download plugin in addition to the vstore download handler
- Fixed  also an issue with wrong download urls for the files mentioned in the  product details view in the Downlods (now Files) section. As this files  are not the files to purchase, there is no need to check if the item is  purchased.

**Order confirmation template**
Moved order confirmation page into a template.

CaMer0n commented on e_url.php:
```diff
@@ -22,9 +22,9 @@ function config($pofile=null)
+			'regex'			=> '^{alias}\/dl/([\d]*)$',
 			'redirect'		=> '{e_PLUGIN}vstore/vstore.php?download=$1&',
-			'sef'			=> '{alias}/download/{item_id}'
+			'sef'			=> '{alias}/dl/{item_id}'
```
How about 'request' (as used in other parts of e107)  or 'get' instead of 'dl' ?

SimSync replied:
Good idea! I'll change it to 'request'.

SimSync commented:
Besides of the updated 'e_url.php', i've also added a css fix for the .fade class in boostrap 4 themes (as mentioned by Jimmi08 here https://github.com/e107inc/e107/issues/2898.
Fixed also an issue with wrong download urls for the files mentioned in the product details view in the Downlods (now Files) section. As this files are not the files to purchase, there is no need to check if the item is purchased.

CaMer0n commented:
Excellent! :+1: Thank you!!
